app/main.py

```
@@ -1,3 +1,44 @@
+from __future__ import annotations
+
 class Distance:
-    # Write your code here
-    pass
+    km: int | float
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def _get_value(self, other: Distance | int | float) -> int | float:
+        return other.km if isinstance(other, Distance) else other
```

Using a helper function like `_get_value` is a good way to avoid repeating code. However, arithmetic operations (`__add__`) and rich comparison operations (`__lt__`, `__gt__`, etc.) should handle unsupported types differently. Rich comparisons should return `NotImplemented` (as per the linked magic methods guide), while arithmetic operations should raise a `TypeError`. A single helper makes this difficult to implement correctly. For instance, `Distance(10) < 'a'` will currently raise a `TypeError` instead of returning `NotImplemented`. It's better to implement the logic within each dunder method to handle these cases correctly.